docs/DeveloperGuide.md

```diff
@@ -284,66 +284,60 @@ show feedback to user.
 
 #### Design Considerations
 
-**Aspect: How add schedule executes:**
-* **Alternative 1 (current choice) Add Schedule into one UnqiueScheduleList**
-    * Pros: All the unique schedules are added into one list, which makes it easier to navigate. It also makes the code look cleaner and more understandable. This is also consistent with the other data classes which also use an underlying list implementation.
-    * Cons: Any iteration will always be O(n) since sorting and checking if there are clashes in `Schedule` happens in this `UniqueScheduleList`
-* **Alternative 2 use HashMap<Date, ScheduleList>**
-    * Pros: Operation does not need to take O(n) time when it comes to checking if the `Event` clashes, since we only check if there are clashes on that particular given date.
-    * Cons: Decreases overall code consistency.
+Since the architecture follows a Model-view-controller design pattern, `FilteredList<data type>` wraps around an `ObservableList<data type>`, the latter of which is used to store in-memory data of `data type` objects. On the front-end, our Ui constantly listens for changes triggered by `FilterCommand#execute` which updates the `FilteredList<data type>` after filtering based on the specified `Predicate`, which then reflects the newly filtered information on the Ui to the user. This design provides a clean way for us to filter data.
+
 
-&nbsp;
 
-### Finding and Editing existing Events
+### Adding Tags to Entries
 
 #### Implementation
 
-The proposed feature of finding and editing existing `Events` is parsed by `FindScheduleCommandParser` and `EditScheduleCommandParser` respectively. Once parsed, those command will be executed by the `FindScheduleCommand` and `EditScheduleCommand` respectively. 
+Tags can be added to the entries on the `schedule`, `contacts` and `todos`  tabs.
+
+These tags have lists that hold `Schedule`, `Person` and `Todo` classes respectively. Let us refer to these as `data type`s. Each of these has a model that encapsulates information for multiple fields. For example, the `Person` model has the fields `Name`, `Relationship`, `Email` etc. that holds the information for each `Person` entry. In addition to these fields, each of these models also has a field called `tags` that tracks the tags that have been applied to that entry. 
```

Comment:
These tags have lists that hold `Schedule`, `Person` and `Todo` classes respectively.

Wait wait I don't think this is applicable for Todos. I'm not sure if the others did this though.
Tags is just an attribute of Todo.

Reply:
bruh my bad its a typo, its supposed to be these tabs